### ISSUE TITLE
xcode-locator outputs invalid JSON

### ISSUE THREAD

**Issue by goodspark** [bazelbuild/bazel#13067]
### Description of the problem / feature request:

xcode-locator outputs invalid JSON

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

- Run `bzl build @bazel_tools//tools/osx:xcode-locator-genrule`
- Then `(path to bazel bin dir)/external/bazel_tools/tools/osx/xcode-locator`

### What operating system are you running Bazel on?

macOS Catalina (10.15.7)

### What's the output of `bazel info release`?

release 3.4.1

### Any other information, logs, or outputs that you want to share?

Example output:

```term
{
	"12.2.0": "/Users/me/Applications/Xcode-12.2.0.12B45b.app/Contents/Developer",
	"12.2": "/Users/me/Applications/Xcode-12.2.0.12B45b.app/Contents/Developer",
	"12": "/Users/me/Applications/Xcode-12.2.0.12B45b.app/Contents/Developer",
	"12.2.0.12B45b": "/Users/me/Applications/Xcode-12.2.0.12B45b.app/Contents/Developer",
}
```

Notice the comma for the last key-value pair in the map. That shouldn't be there for valid JSON.

Basically, the fprintf call here should be a bit smarter.
https://github.com/bazelbuild/bazel/blob/3d78dc7e3a927612c23a6d1542bc38aad5a9703e/tools/osx/xcode_locator.m#L236-L240

**Comment by aiuto** [comment]
@susinmotion  Can you take a look at the fix?
